src/main/java/net/dv8tion/jda/internal/entities/MemberImpl.java

```diff
@@ -207,6 +206,24 @@ public List<Role> getRoles()
         return Collections.unmodifiableList(roleList);
     }
 
+    @Override
+    public boolean hasRoles(@Nonnull Role... roles)
+    {
+        switch (roles.length)
```

Comment:
Imo there is no need for this switch statement. Just do a fast return false if `this.roles.isEmpty()` and otherwise do the contains all check